src/loop_util/loop_util.c

```diff
@@ -0,0 +1,137 @@
+
+        fd = open(image_path, O_CLOEXEC|O_RDWR|O_NONBLOCK|O_NOCTTY);
+        if (fd < 0) {
+                perror("Failed to open file descriptor for provided image");
+                return -1;
+        }
+
+        control = open("/dev/loop-control", O_RDWR|O_CLOEXEC|O_NOCTTY|O_NONBLOCK);
+        if (control < 0) {
+                perror("Failed to open /dev/loop-control");
+                return -1;
+        }
+
+        nr = ioctl(control, LOOP_CTL_GET_FREE);
+        if (nr < 0) {
+                perror("Failed to allocate loop device node");
+                return -1;
+        }
+
+
+        if (asprintf(&loopdev, "/dev/loop%i", nr) < 0) {
+                perror("Failed to retrieve path of loop device node");
+                return -1;
+        }
+
+        printf("initializing loop device node at /dev/loop%d \n", nr);
+
+        loop = open(loopdev, O_CLOEXEC|O_RDWR|O_NONBLOCK|O_NOCTTY);
+        if (loop < 0) {
+                perror("Failed to open loop device loopdev");
+                return -1;
+        }
+
+        if (ioctl(loop, LOOP_SET_FD, fd) < 0) {
+                perror("Failed to set loopback file descriptor fd on loopdev");
+                return -1;
+        }
+
+        loopdev = NULL;
+
+        return nr;
+
+}
+
+static int single_mount(const char *image_path, const char *target) {
+
+
+        int rc;
+        struct libmnt_context *cxt;
+        cxt = mnt_new_context();
+        char *source = NULL;
+
+        if ( asprintf(&source, "/dev/loop%i", get_node(image_path, source)) < 0 ) {
+                perror("Failed to initialize loop device node for mounting");
+                return -1;
+        }
+
+        puts(image_path);
+
+        if ( mnt_context_set_source(cxt, source) < 0 ) {
+                perror("Failed to set mount image source");
+                return -1; 
+        }
+        
+        if ( mnt_context_set_target(cxt, target) < 0 ) {
+                perror("Failed to set mount target directory");
+                return -1;
+        }
+
+        rc = mnt_context_mount(cxt);
+        if (rc) {
+                if (rc > 0) {
+                        perror("Failure to execute mount. Mount(2) error in syscall");
+                        return -1;
+                }
+                else {
+                        perror("Failure to execute mount. Error not related to Mount(2)");
+                        return -1; 
+                }
+        }
+        
+        mnt_free_context(cxt);
+        
+        source = NULL;
+
+        return rc;
+}
+
+
+int main(int argc, char *argv[]) {
+
+        if (!(argc == 3)) {
+                perror("Invalid usage of loop_util. Please input <SOURCE> <TARGET>");
+                return -1;
+        }
+
+        const char *path = argv[1];
+        const char *target = argv[2];
+        int p = single_mount(path, target);
+        if (p < 0)
+                perror("Failure to execute single_mount");
```

Comment:
Don't use perror here since you don't know if `errno` contains an error or not, just fprintf(stderr....). Using errno so late can lead to messages like "Failure to execute single_mount: Success" which is a wee bit confusing